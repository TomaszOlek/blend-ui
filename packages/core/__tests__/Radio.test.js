import React, { createRef } from "react";
import renderer from "react-test-renderer";
import ThemeProvider from "../src/theme/ThemeProvider";
import Label from "../src/Label";
import Radio from "../src/Radio";
import styled from "styled-components";
import { action } from "@storybook/addon-actions";

const LabelText = styled.span`
  vertical-align: middle;
  margin-left: 8px;
`;

const checkAction = e => {
  console.log("CLICK ", e, e.target);
  action(`${e.target.id} was clicked`)(e.target.value, e.target.checked);
};

test("Link renders correctly", () => {
  const tree = renderer
    .create(
      <ThemeProvider>
        <Label fontSize="14px" onClick={checkAction}>
          <Radio
            checked
            value="TABLE"
            onChange={e => {
              console.log(e);
            }}
          />
          <LabelText>selected</LabelText>
        </Label>
        <Label fontSize="14px" onClick={checkAction}>
          <Radio
            value="OBJECT"
            onChange={e => {
              console.log(e);
            }}
          />
          <LabelText>not selected</LabelText>
        </Label>
        <Label fontSize="14px">
          <Radio
            disabled
            onChange={e => {
              console.log(e);
            }}
          />
          <LabelText>disabled</LabelText>
        </Label>
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
