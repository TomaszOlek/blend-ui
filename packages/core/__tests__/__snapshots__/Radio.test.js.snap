// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1w52cfi-0 fMEevm"
>
  <label
    className="Label__LabelElement-sc-1r8ik4v-0 bLjjgM"
    fontSize="14px"
    onClick={[Function]}
  >
    <label
      className="Label__LabelElement-sc-1r8ik4v-0 cGWEb"
      onChange={[Function]}
    >
      <div
        checked={true}
        className="Radio__RadioWrap-sc-mwkpiy-0 fknoAu"
      >
        <input
          className="Radio__RadioInput-sc-mwkpiy-1 cMdnQa"
          id="blend-radio-1"
          type="radio"
          value="TABLE"
        />
        <svg
          aria-hidden={true}
          className="BlendIcon-sc-1i0m6ln-0 kLCpLN"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path fill=\\"currentColor\\" d=\\"M12 5c-3.859 0-7 3.141-7 7s3.141 7 7 7s7-3.141 7-7s-3.141-7-7-7zm0 12c-2.757 0-5-2.243-5-5s2.243-5 5-5s5 2.243 5 5s-2.243 5-5 5z\\"/><path fill=\\"currentColor\\" d=\\"M12 9c-1.627 0-3 1.373-3 3s1.373 3 3 3s3-1.373 3-3s-1.373-3-3-3z\\"/>",
            }
          }
          height="2.0rem"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style={
            Object {
              "color": "#00847A",
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 24 24"
          width="2.0rem"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </div>
      <span
        className="Radio__LabelText-sc-mwkpiy-2 gvyAXi"
      />
    </label>
    <span
      className="Radiotest__LabelText-sc-m692a0-0 iztavn"
    >
      selected
    </span>
  </label>
  <label
    className="Label__LabelElement-sc-1r8ik4v-0 bLjjgM"
    fontSize="14px"
    onClick={[Function]}
  >
    <label
      className="Label__LabelElement-sc-1r8ik4v-0 cGWEb"
      onChange={[Function]}
    >
      <div
        className="Radio__RadioWrap-sc-mwkpiy-0 fknoAu"
      >
        <input
          className="Radio__RadioInput-sc-mwkpiy-1 cMdnQa"
          id="blend-radio-2"
          type="radio"
          value="OBJECT"
        />
        <svg
          aria-hidden={true}
          className="BlendIcon-sc-1i0m6ln-0 kLCpLN"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path fill=\\"currentColor\\" d=\\"M5 12c0 3.859 3.14 7 7 7c3.859 0 7-3.141 7-7s-3.141-7-7-7c-3.86 0-7 3.141-7 7zm12 0c0 2.757-2.243 5-5 5s-5-2.243-5-5s2.243-5 5-5s5 2.243 5 5z\\"/>",
            }
          }
          height="2.0rem"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style={
            Object {
              "color": "#00847A",
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 24 24"
          width="2.0rem"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </div>
      <span
        className="Radio__LabelText-sc-mwkpiy-2 gvyAXi"
      />
    </label>
    <span
      className="Radiotest__LabelText-sc-m692a0-0 iztavn"
    >
      not selected
    </span>
  </label>
  <label
    className="Label__LabelElement-sc-1r8ik4v-0 bLjjgM"
    fontSize="14px"
  >
    <label
      className="Label__LabelElement-sc-1r8ik4v-0 cGWEb"
      onChange={[Function]}
    >
      <div
        className="Radio__RadioWrap-sc-mwkpiy-0 fknoAu"
        disabled={true}
      >
        <input
          className="Radio__RadioInput-sc-mwkpiy-1 cMdnQa"
          id="blend-radio-3"
          type="radio"
        />
        <svg
          aria-hidden={true}
          className="BlendIcon-sc-1i0m6ln-0 kLCpLN"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<path fill=\\"currentColor\\" d=\\"M5 12c0 3.859 3.14 7 7 7c3.859 0 7-3.141 7-7s-3.141-7-7-7c-3.86 0-7 3.141-7 7zm12 0c0 2.757-2.243 5-5 5s-5-2.243-5-5s2.243-5 5-5s5 2.243 5 5z\\"/>",
            }
          }
          height="2.0rem"
          preserveAspectRatio="xMidYMid meet"
          role="img"
          style={
            Object {
              "color": "#00847A",
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 24 24"
          width="2.0rem"
          xmlns="http://www.w3.org/2000/svg"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </div>
      <span
        className="Radio__LabelText-sc-mwkpiy-2 gvyAXi"
      />
    </label>
    <span
      className="Radiotest__LabelText-sc-m692a0-0 iztavn"
    >
      disabled
    </span>
  </label>
</div>
`;
